core/common/src/main/java/alluxio/util/EnvironmentUtils.java

```diff
@@ -45,5 +59,197 @@ public static boolean isKubernetes() {
     return System.getenv("KUBERNETES_SERVICE_HOST") != null;
   }
 
+  /**
+   * Utility to detect the EC2 deployment environment.
+   *
+   * @return true, if running on EC2
+   */
+  public static boolean isEC2() {
+    return isEC2WithUUID() || isEC2WithProductUUID() || isEC2WithInstanceIdentity();
+  }
+
+  /**
+   * Utility to detect the Google compute engine deployment environment.
+   *
+   * @return true, if running on gce
+   */
+  public static boolean isGoogleComputeEngine() {
+    return isGCEWithMetadata() || isGCEWithBiosVendor();
+  }
+
+  /**
+   * Gets the EC2 product code if any.
+   *
+   * @return the first product code if any, an empty string otherwise
+   */
+  public static String getEC2ProductCode() {
+    try {
+      List<String> productCodes = EC2MetadataUtils.getProductCodes();
+      if (productCodes.size() < 1) {
+        return "";
+      }
+      return productCodes.get(0);
+    } catch (Throwable t) {
+      // Exceptions are expected if this instance is not EC2 instance
+      // or this EC2 does not have product code
+      return "";
+    }
+  }
+
+  /**
+   * @return true if this instance is launched from CFT, false otherwise
+   */
+  public static boolean isCFT() {
+    try {
+      String userData = EC2MetadataUtils.getUserData();
```

Comment:
make this call in `UpdateCheck` instance so that we query the metadata service only once and not for each call like isCFT  / isEMR? try to minimize the metadata service calls needed

Reply:
Move this call to UpdateCheck, thanks